Python/specialize.c

```diff
@@ -276,6 +276,7 @@ static int compare_masks[] = {
 void
 _PyCode_Quicken(PyCodeObject *code)
 {
+    #if ENABLE_SPECIALIZATION
```

Comment:
I think this change isn't needed unless you also want to disable superinstructions. That seems like something separate, and less useful, than what you're proposing here.

Reply:
IIRC it had something to do with making the interpreter easier to debug when *nothing* works. The more you can rule out the less code you have left to debug.

Reply:
> I wonder if the changes in `ceval.c` would be cleaner if we just changed the "ADAPTIVE_COUNTER" macro definitions when `ENABLE_SPECIALIZATION` is zero (instead of sprinkling these `#if`s everywhere). Might be about the same level of complexity, though.

Possibly. But if we want to exclude this code from compilation we would need to rewrite ADAPTIVE_COUNTER_IS_ZERO so that it's not in an if condition.

Reply:
> I think this change isn't needed unless you also want to disable superinstructions. That seems like something separate, and less useful, than what you're proposing here.

We could have a separate flat ENABLE_SUPERINSTRUCTIONS for that. We could have toggles for different kinds of optimisations, with a global one to turn them all off for debugging (the individual ones can be used to assess the performance gain of the parts, and to better document what is what):

ENABLE_BYTECODE_OPTIMIZATIONS = True
ENABLE_SPECIALIZATION =  ENABLE_BYTECODE_OPTIMIZATIONS and True
ENABLE_SUPERINSTRUCTIONS =  ENABLE_BYTECODE_OPTIMIZATIONS and True
ENABLE_PEEPHOLER =  ENABLE_BYTECODE_OPTIMIZATIONS and True
...

Reply:
This is a feature for extreme debugging. I'd keep it as simple as possible.

Reply:
I'll merge then. We can iterate on this.